client/src/ctx.rs

```diff
@@ -325,18 +326,44 @@ impl SessionContext {
 
     pub async fn write_line_protocol_file(&self, path: impl AsRef<Path>) -> Result<ResultSet> {
         let file = std::fs::File::open(path)?;
-        let reader = std::io::BufReader::new(file);
-        let mut batch = Vec::new();
-        for (i, line) in reader.lines().enumerate() {
-            if i % 10000 == 0 && !batch.is_empty() {
-                self.write_line_protocol(batch.join("\n").into()).await?;
-                batch.clear();
+        let size = file.metadata()?.len();
+        let mut reader = std::io::BufReader::new(file);
+        let pb = progress_bar::new_with_size(size);
+
+        let (tx, mut rx) = mpsc::channel::<String>(10);
+
+        let producer = tokio::spawn(async move {
+            let mut buffer = String::new();
+            let mut line_count = 0;
+            while let Ok(size) = reader.read_line(&mut buffer) {
+                if size == 0 {
```

Comment:
Why not put the check to see if the file is empty right after the file is opened?

```rust
let size = file.metadata()?.len();
if size == 0 {
    return Ok(ResultSet::Bytes(("".into(), 0)));
}
```